ci.py

```diff
@@ -35,37 +46,83 @@ def create_parser() -> argparse.ArgumentParser:
-    subprocess.run(list_command, check=True)
-    subprocess.run(version_command, env=env_with_pantsd, check=True)
-    subprocess.run(list_command, env=env_with_pantsd, check=True)
+    with setup_python_version(test_python_version):
+      subprocess.run(version_command, check=True)
+      subprocess.run(list_command, check=True)
+      subprocess.run(version_command, env=env_with_pantsd, check=True)
+      subprocess.run(list_command, env=env_with_pantsd, check=True)
 
 
 @contextmanager
 def setup_pants_version(test_pants_version: PantsVersion):
   """Modify pants.ini to allow the pants version to be unspecified or keep what was originally there."""
-  with open(PANTS_INI, 'r') as f:
-    original_pants_ini = list(f.readlines())
-  pants_version_already_specified = any(line.startswith("pants_version:") for line in original_pants_ini)
-  if test_pants_version == PantsVersion.config and not pants_version_already_specified:
-    raise ValueError("You requested to use the pants_version from pants.ini for this test, but pants.ini "
-                     "does not include a pants_version! Please update pants.ini and run again.")
-  if test_pants_version == PantsVersion.unspecified and pants_version_already_specified:
-    with open(PANTS_INI, 'w') as f:
-      # NB: we must not only remove the original definition of `pants_version`, but also
-      # any lines that make use of it, such as contrib packages pinning their version to `pants_version`.
-      f.writelines(line for line in original_pants_ini if "pants_version" not in line)
+  original_lines = read_pants_ini()
+  if test_pants_version == PantsVersion.unspecified:
+    write_config_entry(entry_name="pants_version", entry_value=None)
+  elif test_pants_version == PantsVersion.config:
+    pants_version_defined = any(line.startswith("pants_version") for line in original_lines)
+    if not pants_version_defined:
+      raise ValueError("You requested to use the pants_version from pants.ini for this test, but pants.ini "
+                       "does not include a pants_version! Please update pants.ini and run again.")
   yield
+  write_pants_ini(original_lines)
+
+
+@contextmanager
+def setup_python_version(test_python_version: PythonVersion):
+  """Modify pants.ini to allow the Python version to be unspecified or change to what was requested."""
+  original_lines = read_pants_ini()
+  requested_python_version = None if test_python_version == PythonVersion.unspecified else test_python_version.value
+  write_config_entry(entry_name="pants_runtime_python_version", entry_value=requested_python_version)
+  yield
+  write_pants_ini(original_lines)
+
+
+def read_pants_ini() -> List[str]:
+  with open(PANTS_INI, 'r') as f:
+    return list(f.readlines())
+
+
+def write_pants_ini(lines: Iterable[str]) -> None:
   with open(PANTS_INI, 'w') as f:
-    f.writelines(original_pants_ini)
+    f.writelines(lines)
+
+
+def write_config_entry(*, entry_name: str, entry_value: Optional[str]) -> None:
```

Comment:
Is there a reason this doesn't use [`configparser`](https://docs.python.org/3.7/library/configparser.html)?